src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -215,29 +167,100 @@ export class MeetingManager implements AudioVideoObserver {
     this.audioVideoObservers = {};
   }
 
-  async join({
-    meetingInfo,
-    attendeeInfo,
-    deviceLabels = DeviceLabels.AudioAndVideo,
-    eventController,
-    meetingManagerConfig,
-  }: MeetingJoinData) {
-    if (meetingManagerConfig) {
-      this.meetingManagerConfig = meetingManagerConfig;
-    }
-    this.configuration = new MeetingSessionConfiguration(
-      meetingInfo,
-      attendeeInfo
+  async join(
+    meetingSessionConfiguration: MeetingSessionConfiguration,
+    options?: MeetingManagerJoinOptions
+  ): Promise<void> {
+    const {
+      deviceLabels,
+      eventController,
+      logger,
+      enableWebAudio,
+      activeSpeakerPolicy,
+    } = this.parseJoinParams(meetingSessionConfiguration, options);
+    this.meetingSessionConfiguration = meetingSessionConfiguration;
+    this.meetingId = this.meetingSessionConfiguration.meetingId;
+    this.logger = logger;
+
+    const deviceController = new DefaultDeviceController(logger, {
+      enableWebAudio: enableWebAudio,
+    });
+    this.meetingSession = new DefaultMeetingSession(
+      meetingSessionConfiguration,
+      logger,
+      deviceController,
+      eventController,
     );
 
-    this.meetingRegion = meetingInfo.MediaRegion;
-    this.meetingId = this.configuration.meetingId;
+    this.audioVideo = this.meetingSession.audioVideo;
 
     if (eventController) {
-      this.eventController = eventController;
+      eventController.addObserver(this.eventDidReceiveRef);
+    } else {
+      this.meetingSession.eventController.addObserver(this.eventDidReceiveRef);
+    }
+
+    this.setupAudioVideoObservers();
+    this.setupDeviceLabelTrigger(deviceLabels);
+    await this.listAndSelectDevices(deviceLabels);
+    this.publishAudioVideo();
+    this.setupActiveSpeakerDetection(activeSpeakerPolicy);
+    this.meetingStatus = MeetingStatus.Loading;
+    this.publishMeetingStatus();
+  }
+
+  private parseJoinParams(
+    meetingSessionConfiguration: MeetingSessionConfiguration,
+    options: MeetingManagerJoinOptions | undefined
```

Comment:
Can make `options` optional as well like the `join` method and drop `| undefined`.